csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest.sh

```diff
@@ -18,8 +18,12 @@ echo "Current NuGet package version is $CurrentOnnxRuntimeVersion"
 
 if [ $RunTestCsharp = "true" ]; then
   if [[ $IsMacOS == "True" || $IsMacOS == "true" ]]; then
-    mkdir -p $BUILD_BINARIESDIRECTORY/models
-    ln -s $BUILD_SOURCESDIRECTORY/cmake/external/onnx/onnx/backend/test/data/node $BUILD_BINARIESDIRECTORY/models/opset16
+    # TODO(#12040): The test should figure out the opset version from the model file. Remove it from the path.
+    ONNX_DIR="${BUILD_SOURCESDIRECTORY}/cmake/external/onnx"
+    ONNX_VERSION_NUMBER=$(cat "${ONNX_DIR}/VERSION_NUMBER" | sed -E 's/([0-9]+\.[0-9]+\.[0-9]+).*/\1/')
```

Comment:
I have a concern that VERSION_NUMBER from ONNX repo can be something like 1.12.0rc1 during package verification and further getting OPSET_VERSION will fail. If we are confident that we only update the script here after having ONNX released, please ignore this comment.

Otherwise, perhaps grep this opset version from [here](https://github.com/onnx/onnx/blob/3428633b071c7c5b47c60414d571077f2aebcaa1/onnx/defs/schema.h#L922) would be easier and more reliable.

Reply:
The sed command handles `1.12.0.rc1`:
```sh
$ ONNX_VERSION_NUMBER=$(echo "1.12.0rc1" | sed -E 's/([0-9]+\.[0-9]+\.[0-9]+).*/\1/')
$ echo $ONNX_VERSION_NUMBER
1.12.0
```